### PR TITLE
Throttle instead of debouncing scroll events

### PR DESCRIPTION
**Pull request by pcurrivan**
Currently, scroll events cause a render after no scroll events occur for
50ms.  This causes the viewport to appear empty while scrolling quickly.
This commit changes it so that a render occurs every 50ms while
scrolling.  (The viewport may still appear empty sometimes if rendering
takes more than 50ms)

### CODE REVIEW

File: slick.grid.js [hunk 1]
```diff
@@ -2412,18 +2415,19 @@ if (typeof Slick === "undefined") {
...
-              Math.abs(lastRenderedScrollTop - scrollTop) < viewportH &&
-              Math.abs(lastRenderedScrollLeft - scrollLeft) < viewportW)) {
+        var dx = Math.abs(lastRenderedScrollLeft - scrollLeft);
+        var dy = Math.abs(lastRenderedScrollTop - scrollTop);
+
+        if (dx > 20 || dy > 20) {
+
+          // if rendering is forced or scrolling is small enough to be "easy", just render
+          if (options.forceSyncScrolling || (dy < viewportH && dx < viewportW)) {
             render();
```
Comment (vlsi) [inline slick.grid.js]: I wonder if `requestAnimationFrame` should be added here (and to `scrollThrottle` as well)

Comment (pcurrivan) [reply slick.grid.js]: Good question.  I don't know if or how that function should be used.  There are performance issues (see my comment below).

**Comment by pcurrivan** [comment]
I will say that on Chrome, my method here works pretty well when scrolling by dragging the scroll bar.  That is why I optimistically opened this pull request.  However, the situation is more complicated.  Scrolling quickly with a trackpad on Chrome still shows an empty viewport.  On Firefox (quantum), scrolling by dragging or with a trackpad both still show an empty viewport.  Waterfox (old firefox engine), is just choppy and slow.  My conclusion is that my solution here is not adequate for all the different browsers, and I don't have time at the moment to figure it out, so closing this for now.

**Comment by 6pac** [comment]
It was a noble attempt. I implemented this on a branch locally this morning, and in chrome I couldn't actually see any  difference with or without the change! Someone has noted that Ag-grid does this, so perhaps it's worth a peek at their source?

**Pull request closed without merging**